GitExtUtils/Validates.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft
```

Comment:
Have you chosen this namespace by intention?
```suggestion
namespace GitExtUtils
```

Reply:
Microsoft.Validates replaces Microsoft.Assumes so changing the namespace would require a manual change in every file using it.
Microsoft.Assumes should change the exception message (see discussion above) as it is used in other products, then we could switch back to Assumes. 
But it should be commented at least.

Reply:
I can add a comment to explain the namespace choice.

Off topic, but: has there been any discussion about consolidating the namespaces and assembly names used across Git Extensions? They're fairly inconsistent.

Reply:
Added comment and fixed merge conflicts.

Reply:
> Off topic, but: has there been any discussion about consolidating the namespaces and assembly names used across Git Extensions? They're fairly inconsistent.

Not what I have seen